ui/page.go

```diff
@@ -145,6 +145,18 @@ func (page pageCommon) Layout(gtx layout.Context, body layout.Widget) layout.Dim
 	})
 }
 
+func (page pageCommon) LayoutWithModal(gtx layout.Context, body layout.Widget, modal layout.Widget) layout.Dimensions {
```

Comment:
This can just be named "Modal" and you don't need to specify its second parameter as a widget, it can just be dimension. There's no layout there, all you need are the dimensions.